tests/core/server/test_rate_limits.py

```diff
@@ -217,3 +217,49 @@ async def test_percentage_limits(self):
             if not response:
                 saw_disconnect = True
         assert saw_disconnect
+
+    @pytest.mark.asyncio
+    async def test_too_many_outgoing_messages(self):
```

Comment:
these two tests demonstrate the problem. Prior to this patch, we would have the "incoming" behavior in both cases, both incoming and outgoing messages.